### PR TITLE
Net 202 

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
added endpoint to netclient server to permit GUI to:
1. join via basic auth
2. get auth endpoint for SSO

## Provide link to Netmaker PR if required
N/A
## Provide testing steps
NET-202

### CODE REVIEW

File: functions/httpserver.go [hunk 1]
```diff
@@ -211,5 +216,96 @@ func nodePeers(c *gin.Context) {
...
+			c.JSON(http.StatusOK, gin.H{"authendpoint": string(msg)})
+			return
+		}
+	}
+}
+
+func hostForSSO() models.Host {
+	host := config.Netclient()
+	ip, err := getInterfaces()
+	if err != nil {
```
Comment (abhishek9686) [inline functions/httpserver.go:296]: why is `getInterfaces()` called here, how is this related to SSO?

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

**Review by Aceix — APPROVED** [review]
lgtm. will test again when connecting the gui

File: functions/httpserver.go (hunk 1 above)
Comment (mattkasun) [reply functions/httpserver.go:296]: following the pattern from logic.Register()

**Review by TobiaszCudnik — APPROVED** [review]

**Review by abhishek9686 — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
pr passes
critical list passes
